### PR TITLE
Add linear regression demo code

### PR DESCRIPTION
**Pull request by ecwu**
This code is a demo code for Linear regression, including two regression method:

- Gradient Descent
- Normal Equation

No package like `Scikit-learn` is used, Pure `numpy`. 😄

### CODE REVIEW

**Review by TerenceLiu98 — APPROVED** [review]
thanks for helping
